src/processor.cpp

```diff
@@ -0,0 +1,394 @@
+#include "game.h"
+#include "map.h"
+#include "mapdata.h"
+#include "generic_factory.h"
+#include "messages.h"
+#include "json.h"
+#include "vehicle.h"
+#include "processor.h"
+
+#include <string>
+
+//process namespace
+
+namespace
+{
+    struct process_data;
+    using process_id = string_id<process_data>;
+
+    struct process_data {
+
+        process_data() :
+            was_loaded(false) {};
+
+        process_id id;
+        std::string name;
+        std::vector<std::string> components;
+        std::string output;
+        int fuel_intake;
+        int duration;
+        //TODO: Skill used
+
+        bool was_loaded;
+        void load(JsonObject &jo, const std::string &src);
+        void check() const;
+
+    };
+
+    process_id get_process_id( std::string id )
+    {
+        return process_id( id );
+    }
+
+    generic_factory<process_data> processes_data("process type", "id");
+
+}
+
+void process_data::load(JsonObject &jo, const std::string &)
+{
+    mandatory(jo, was_loaded, "name", name);
+    mandatory(jo, was_loaded, "components", components);
+    mandatory(jo, was_loaded, "output", output);
+    mandatory(jo, was_loaded, "fuel_intake", fuel_intake);
+    mandatory(jo, was_loaded, "duration", duration);
+}
+
+void process_data::check() const
+{
+    for (const std::string &elem : components) {
+        if (!item::type_is_defined( elem )) {
+            debugmsg("Invalid components \"%s\" in \"%s\".", elem.c_str(), id.c_str());
+        }
+    }
+    if (!item::type_is_defined( output )) {
+        debugmsg("Invalid output \"%s\" in \"%s\".", output.c_str(), id.c_str());
+    }
+    if ( !(fuel_intake >= 0) ) {
+        debugmsg("Invalid fuel intake \"%d\" in \"%s\".", fuel_intake, id.c_str());
+    }
+    if (!(duration > 0)) {
+        debugmsg("Invalid duration \"%d\" in \"%s\".", duration, id.c_str());
+    }
+}
+
+void processes::load(JsonObject &jo, const std::string &src)
+{
+    processes_data.load(jo, src);
+}
+
+void processes::check()
+{
+    processes_data.check();
+}
+
+void processes::reset()
+{
+    processes_data.reset();
+}
+
+// Processor namespace
+
+namespace
+{
+struct processor_data;
+using processor_id = string_id<processor_data>;
+
+struct processor_data {
+
+    processor_data() :
+        was_loaded( false ) {};
+
+    processor_id id; //furniture
+    furn_str_id next_type;
+    itype_id fuel_type;
+
+
+    std::vector<std::string> processes;
```

Comment:
Use the process id here. Avoid storing raw strings as this makes using them less straightforward.
The json readers support reading into ids.